src/main/java/com/minecolonies/coremod/client/gui/WindowScan.java

```diff
@@ -0,0 +1,318 @@
+    private final TextField pos1y;
+    private final TextField pos1z;
+
+    /**
+     * Pos 2 text fields.
+     */
+    private final TextField pos2x;
+    private final TextField pos2y;
+    private final TextField pos2z;
+
+    /**
+     * Constructor for when the player wants to scan something.
+     * @param pos1 the first pos.
+     * @param pos2 the second pos.
+     */
+    public WindowScan(final BlockPos pos1, final BlockPos pos2)
+    {
+        super(Constants.MOD_ID + BUILDING_NAME_RESOURCE_SUFFIX);
+        this.pos1 = pos1;
+        this.pos2 = pos2;
+        registerButton(BUTTON_CONFIRM, this::confirmClicked);
+        registerButton(BUTTON_CANCEL, this::discardClicked);
+        registerButton(BUTTON_SHOW_RES, this::updateResources);
+
+        pos1x = findPaneOfTypeByID(POS1X_LABEL, TextField.class);
+        pos1y = findPaneOfTypeByID(POS1Y_LABEL, TextField.class);
+        pos1z = findPaneOfTypeByID(POS1Z_LABEL, TextField.class);
+
+        pos2x = findPaneOfTypeByID(POS2X_LABEL, TextField.class);
+        pos2y = findPaneOfTypeByID(POS2Y_LABEL, TextField.class);
+        pos2z = findPaneOfTypeByID(POS2Z_LABEL, TextField.class);
+    }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
+
+        if (!Minecraft.getMinecraft().player.capabilities.isCreativeMode)
+        {
+            pos1x.disable();
+            pos1y.disable();
+            pos1z.disable();
+
+            pos2x.disable();
+            pos2y.disable();
+            pos2z.disable();
+        }
+
+        pos1x.setText(String.valueOf(pos1.getX()));
+        pos1y.setText(String.valueOf(pos1.getY()));
+        pos1z.setText(String.valueOf(pos1.getZ()));
+
+        pos2x.setText(String.valueOf(pos2.getX()));
+        pos2y.setText(String.valueOf(pos2.getY()));
+        pos2z.setText(String.valueOf(pos2.getZ()));
+    }
+
+    /**
+     * On cancel button.
+     */
+    private void discardClicked()
+    {
+        close();
+    }
+
+    /**
+     * On confirm button.
+     */
+    private void confirmClicked()
+    {
+        final String name = findPaneOfTypeByID(NAME_LABEL, TextField.class).getText();
+
+        final int x1 = Integer.parseInt(pos1x.getText());
+        final int y1 = Integer.parseInt(pos1y.getText());
+        final int z1 = Integer.parseInt(pos1z.getText());
+
+        final int x2 = Integer.parseInt(pos2x.getText());
+        final int y2 = Integer.parseInt(pos2y.getText());
+        final int z2 = Integer.parseInt(pos2z.getText());
+
+        MineColonies.getNetwork().sendToServer(new ScanOnServerMessage(new BlockPos(x1, y1, z1), new BlockPos(x2, y2, z2), name));
+        close();
+    }
+
+    @Override
+    public boolean onKeyTyped(final char ch, final int key)
+    {
+        if (key == ENTER_KEY)
+        {
+            updateResources();
+        }
+        return super.onKeyTyped(ch, key);
+    }
+
+    /**
+     * Clears and resets/updates all resources.
+     */
+    private void updateResources()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 47 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)